src/models/real-fevr-marketplace.ts

```diff
@@ -0,0 +1,128 @@
+import {RealFevrMarketplaceMethods} from '@methods/real-fevr-marketplace';
+import {AbiItem} from 'web3-utils';
+import {Errors} from '@interfaces/error-enum';
+import {ERC20} from '@models/erc20';
+import {RealFevrOpener} from '@models/real-fevr-opener';
+import {toSmartContractDecimals} from '@utils/numbers';
+
+export class RealFevrMarketplace extends Model<RealFevrMarketplaceMethods> implements Deployable {
+  constructor(
+    web3Connection: Web3Connection|Web3ConnectionOptions,
+    contractAddress?: string,
+    readonly collectiblesAddress?: string,
+    readonly tokenAddress?: string,
+  ) {
+    super(web3Connection, RealFevrMarketplaceJson.abi as AbiItem[], contractAddress);
+  }
+
+  private _isETHTransaction!: boolean;
+  isETHTransaction() { return this._isETHTransaction; }
+
+  private _erc20!: ERC20;
+  get erc20() { return this._erc20; }
+
+  private _opener!: RealFevrOpener;
+  get opener() { return this._opener; }
+
+  async loadContract() {
+    if (!this.contract)
+      await super.loadContract();
+
+    const tokenAddress = await this.getERC20TokenAddress() || this.tokenAddress;
+    if (!tokenAddress)
+      throw new Error(Errors.MissingERC20AddressOnContract);
+
+    const collectiblesAddress = await this.getERC721TokenAddress() || this.collectiblesAddress;
+    if (!collectiblesAddress)
+      throw new Error(Errors.MissingERC721AddressOnContract);
+
+    this._isETHTransaction = tokenAddress === '0x0000000000000000000000000000000000000000';
+
+    if (!this.isETHTransaction()) {
+      // Set Token Address Contract for easy access
+      this._erc20 = new ERC20(this.web3Connection, tokenAddress);
+      await this._erc20.loadContract();
+    }
+
+    this._opener = new RealFevrOpener(this.web3Connection, collectiblesAddress);
+    await this._opener.loadContract();
+  }
+
+  async start() {
+    await super.start();
+    await this.loadContract();
+  }
+
+  async deployJsonAbi(collectiblesAddress: string, tokenAddress: string = '0x0000000000000000000000000000000000000000') {
+    const deployOptions = {
+        data: RealFevrMarketplaceJson.bytecode,
+        arguments: [tokenAddress, collectiblesAddress]
+    };
+
+    return this.deploy(deployOptions, this.web3Connection.Account);
+  }
+
+  async putERC721OnSale(tokenId: number, price: number) {
+    const valueWithDecimals = toSmartContractDecimals(
+      price,
+      await this.isETHTransaction() ? 18 : this.erc20.decimals,
+    );
+
+    return this.sendTx(
+      this.contract.methods.putERC721OnSale(tokenId, valueWithDecimals as number),
+    );
+  }
+
+  async removeERC721FromSale(tokenId: number) {
+    return this.sendTx(this.contract.methods.removeERC721FromSale(tokenId));
+  }
+
+  async buyERC721(tokenId: number, value: number = 0) {
+    const valueWithDecimals = toSmartContractDecimals(
+      value,
+      await this.isETHTransaction() ? 18 : this.erc20.decimals,
+    );
+
+    return this.sendTx(
```

Comment:
don't break lines for arguments. use line breaks to separate logic or when max-chars-per-line reached

Reply:
done